app/src/hooks/useTokenPrice.tsx

```diff
@@ -0,0 +1,42 @@
+import { useQuery } from '@apollo/react-hooks'
+import { BigNumber, bigNumberify } from 'ethers/utils'
+import gql from 'graphql-tag'
+import { useEffect, useState } from 'react'
+
+import { getLogger } from '../util/logger'
+import { Status } from '../util/types'
+
+const logger = getLogger('useTokenPrice')
+
+const query = gql`
+  query GetTokenPrice($id: String!) {
+    token(id: $id) {
+      usdPerToken
+    }
+  }
+`
+
+type GraphResponse = {
+  tokenPrice: string
+}
+
+export const useTokenPrice = (address: string) => {
+  const [tokenPrice, setTokenPrice] = useState(new BigNumber(0))
+
+  const { data: data, error: error, loading: loading } = useQuery<GraphResponse>(query, {
+    notifyOnNetworkStatusChange: true,
+    skip: false,
+    variables: { id: address },
+  })
+
+  useEffect(() => {
+    if (data && data.tokenPrice) {
+      setTokenPrice(bigNumberify(data.tokenPrice))
```

Comment:
Once the path mentioned above is updated, bigNumberify is freaking out on [this market](http://localhost:3000/#/0xb4277c368b817db02bab8b684c9bfe10200461f7) with an `invalid BigNumber string value` error message, not really sure why 🤔